### PR TITLE
chore(python): Print Python interpreter version

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This relates to https://github.com/GreptimeTeam/greptimedb/discussions/3117 for improving user experience.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Comment by tisonkun** [comment]
@discord9 also I noticed that we have both pyo3 and rustpython for interop. What's the difference and why do we maintain two solutions?

**Review by waynexia — COMMENTED** [review]
Can you also add this for rustpython interpreter? At `src/script/src/python/rspython/copr_impl.rs`

**Comment by tisonkun** [comment]
@waynexia I'm thinking of this. Will take a look. May you take a look at https://github.com/GreptimeTeam/greptimedb/pull/3118#issuecomment-1881153572?

**Comment by waynexia** [comment]
>What's the difference and why do we maintain two solutions?

The original purpose is to support different backends. For an inappropriate example, you can think of this as PG and MySQL protocols.

These two implementations also have some slight differences. Like GIL or runtime requirement. I don't think they are in conflict but a supplement to each other.

**Comment by discord9** [comment]
> @discord9 also I noticed that we have both pyo3 and rustpython for interop. What's the difference and why do we maintain two solutions?

|  Backend  |  RustPython   | CPython  |
|  ----  | ----  | ----  |
| GIL | No | Yes |
| need extra dyn lib(i.e. libpython3.xx) to run  | No | Yes |
|Can use external Python Package(i.e. numpy) | No | Yes |

So the idea is that RustPython can support simple scripting without extra setup, and cpython impl can use external python library

**Comment by tisonkun** [comment]
Thank you @waynexia & @discord9! Updated for RustPython.

File: src/script/src/python/rspython/copr_impl.rs [hunk 1]
```diff
@@ -221,7 +221,14 @@ pub(crate) fn init_interpreter() -> Rc<Interpreter> {
                     init_greptime_builtins("greptime", vm);
                     init_data_frame("data_frame", vm);
                 }));
-                info!("Initialized Python interpreter.");
+                interpreter
+                    .enter(|vm| {
+                        let sys = vm.sys_module.clone();
+                        let version = sys.get_attr("version", vm)?.str(vm)?;
+                        info!("Initialized RustPython interpreter {version}");
```
Comment (tisonkun) [inline src/script/src/python/rspython/copr_impl.rs:228]: FYI -

```
Initialized RustPython interpreter 3.11.0alpha (heads/main:9ed513741, Apr 17 2023, 22:03:55) 
```

File: src/script/src/python/pyo3/utils.rs [hunk 2]
```diff
@@ -45,8 +45,12 @@ pub(crate) fn init_cpython_interpreter() -> PyResult<()> {
     if !*start {
         pyo3::append_to_inittab!(greptime_builtins);
         pyo3::prepare_freethreaded_python();
+        let version = Python::with_gil(|py| -> PyResult<String> {
+            let builtins = PyModule::import(py, "sys")?;
+            builtins.getattr("version")?.extract()
+        })?;
         *start = true;
-        info!("Started CPython Interpreter");
+        info!("Started CPython Interpreter {version}");
```
Comment (tisonkun) [inline src/script/src/python/pyo3/utils.rs:53]: FYI - 
```
Started CPython Interpreter 3.11.6 (main, Nov  2 2023, 04:39:43) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

**Review by waynexia — APPROVED** [review]
LGTM, thanks @tisonkun

**Comment by tisonkun** [comment]
It seems we need one more explicit approval. @discord9 could you tick one?

**Review by discord9 — APPROVED** [review]

**Pull request merged**